docs/metadata.md

```diff
@@ -22,19 +22,25 @@ directories to be created and the directories are populated with files.
 For example, the following userdata file:
 ```JSON
 {
-    "ssh" : {
-        "sshd_config" : {
-            "perm" : "0600",
-            "content": "PermitRootLogin yes\nPasswordAuthentication no"
-        }
-    },
-    "foo" : {
-        "bar" : "foobar",
-        "baz" : {
-            "perm": "0600",
-            "content": "bar"
-        }
+  "ssh": {
+    "entries": {
+      "sshd_config": {
+        "perm": "0600",
+        "content": "PermitRootLogin yes\nPasswordAuthentication no"
+      }
```

Comment:
This turns out not to be backwards compatible (sorry, I somehow convinced myself before that it would be).

I can't see a way to fix that without reintroducing the issue with having directory entries named `perm` etc, so I think we should just bite the bullet, but I leave the final determination on that to @rn .

Reply:
I'm fine with making this change not being backwards compat at this stage. We haven't done a release yet...I'll give a heads up on the community slack